src/pytorch_lightning/strategies/fully_sharded_native.py

```diff
@@ -215,9 +239,21 @@ def setup(self, trainer: "pl.Trainer") -> None:
         assert self.lightning_module is not None
         self.lightning_module._device = self.root_device
 
+        assert isinstance(self.model, pl.LightningModule)
+        self.model = _LightningModuleWrapperBase(self.model)
+        if is_overridden("configure_sharded_model", self.lightning_module):
```

Comment:
I would merge this with the checks for popping "auto_wrap_policy" in `_setup_model`. Either by pulling those to here or moving this into `_setup_model`

Reply:
`auto_wrap_policy` is part of `kwargs` used within the sharding context manager, so we shouldn't do this.